episodes/what-is-an-image.md

```diff
@@ -0,0 +1,540 @@
+8-bit may be a better choice to reduce the required file storage space. As 
+always it's about choosing the best fit for your specific project! 
+
+For more information on bit depths and types - we highly recommend 
+[this chapter of Pete Bankhead's free bioimage book](
+https://bioimagebook.github.io/chapters/1-concepts/3-bit_depths/bit_depths.html).
+
+:::::::::::::::::::::::::::::::::::::: callout
+
+### Clipping and overflow
+
+It's important to be aware of what image type and bit depth you are using. If 
+you try to store values outside of the valid range this can lead to clipping 
+and overflow.
+
+- Clipping: Values outside the valid range are changed to the closest valid 
+value. For example, storing 1000 in a `uint8` image may result in 255 being 
+stored instead (the max value)
+
+- Overflow: For NumPy arrays, values outside the valid range are 
+'wrapped around' to give the new result. For example, storing 256 in a `uint8` 
+image (max 255) would give 0, 257 would give 1 and so on...  
+
+Clipping and overflow result in data loss - you can't get the original values 
+back! So it's always good to keep the data type in mind when doing image 
+processing operations (as we will see in later episodes), and also when 
+converting between different bit depths. 
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Coordinate system
+
+We've seen that images are arrays of numbers with a specific shape (dimensions) 
+and data type. How do we access specific values from this array? What coordinate 
+system is Napari using?
+
+To look into this, let's hover over pixels in our mitosis image and examine the 
+coordinates that appear to the left of the pixel value:
+
+![](fig/coordinates.png){alt="A screenshot of Napari - with the mouse cursor 
+hovering over a pixel and highlighting the corresponding coordinates"}
+
+As you move around, you should see that the lowest coordinate values are at the 
+top left corner, with the first value increasing as you move down and the second 
+value increasing as you move to the right. This is different to the standard 
+coordinate systems you may be used to (for example, from making graphs):
+
+![](fig/coordinate-system.png){alt="Diagram comparing a standard graph 
+coordinate system (left) and the image coordinate system (right)"}
+
+Note that Napari lists coordinates as [y, x] or [rows, columns], so e.g. [1,3] 
+would be the pixel in row 1 and column 3. Remember that these coordinates always 
+start from 0 as you can see in the diagram below:
+
+![](fig/coordinates-on-image.png
+){alt="A diagram showing how pixel coordinates change over a simple 4x4 image" width=50%}
+
+For the mitosis image, these coordinates are in pixels, but we'll see in later 
+episodes that images can also be scaled based on resolution (e.g. in 
+micrometres). Also, bear in mind that images with more dimensions (e.g. a 3D 
+image) will have longer coordinates like [z, y, x]...
```

Comment:
I thought slice was typically stored in the third dimension so it would be [y,x,z] ?

Reply:
I knot that python does it as `z, y, x` but different instruments may have axis stored in different orders. One of the plugin we developed had to manage these changes because the underlying C-code had been written in the axis as you said.

Reply:
Yes python doesn't have very strong conventions for the order of axes, it varies by library/plugin/filetype... It tends to be (z, y, x), but I've added a lot of coverage of this to the 'Multi-dimensional images' episode